test/e2e/framework.go

```diff
@@ -704,8 +711,78 @@ func (data *TestData) deployAntreaFlowExporter(ipfixCollector string) error {
 	return data.mutateAntreaConfigMap(nil, ac, false, true)
 }
 
+// deployFlowVisibilityClickHouse deploys ClickHouse operator and DB.
+func (data *TestData) deployFlowVisibilityClickHouse() (*PodIPs, error) {
```

Comment:
Looking at this code, I feel like we should stop mixing the Flow Aggregator tests with the rest of the Antrea tests.
Looking at the different jobs we have in CI which run on Kind clusters (in different networking configurations), I don't think these jobs should run the Flow Aggregator tests. We should have a dedicated job for Flow Aggregator and ClickHouse testing IMO, and the ClickHouse operator deployment should probably happen outside of the Go test code itself to keep things simple.
The steps would be as follows:
- create a kind cluster
- deploy Antrea (default configuration + flow exporter enabled)
- deploy the Flow Aggregator + IPFIX Collector + ClickHouse
- run Go test cases

@tnqn what do you think? I don't believe it is necessary to run Flow Aggregator tests in all the different configurations we support (Encap, NoEncap, Hybrid, Policies disabled, ProxyAll...), even if there some small differences on the Flow Exporter side (if AntreaProxy is disabled for example).

Reply:
I would agree no need to run aggregator test with all configurations.

Reply:
I agree as well. I think we can start with running it for encap only. What I don't know if how we should adjust test code for deploying flow aggreator and clickhouse only in a single config. Any thought?

Reply:
> What I don't know if how we should adjust test code for deploying flow aggreator and clickhouse only in a single config. Any thought?

These should be deployed by a "cluster creation" script and not as part of the Go test code directly. IMO, we should try to move away (when possible) from the current model which consists in mutating the Antrea config and deploying extra components as part of the test cases themselves.

The test script would deploy the Flow Aggregator and the Clickhouse stuff as part of cluster creation, before running the test cases.

Reply:
Agree that running the FA e2e for all configurations doesn't make much sense. Regarding how we're going to address is, I have a couple of questions:
- Should the test be triggered by adding a comment to PR? and do we need new workflows for it to run?
- for the jenkins e2e, do we still want to include the FA-related tests?
- Is there an example of how this is preferred to be done? 
- depending on the approach there may be a non-trivial amount of refactoring needed. do you think we can add a (temp) additional option to the current e2e test and only run it in certain configuration for this PR?

Reply:
- you can run the tests in a Kind cluster, no PR comment command needed
- let's start with the Kind cluster (now that we use the OVS kernel datapath for Kind tests, it's much easier to run Antrea in Kind that it used to be), and skip the FA tests for `jenkins-e2e`
- I'd recommend looking into adding a script similar to https://github.com/antrea-io/antrea/blob/main/ci/kind/test-e2e-kind.sh, which we can invoke from the Github workflow. It would also invoke `kind-setup.sh` but would also take care of deploying Antrea with the right options, along with the FlowAggregator, Flow Collector and ClickHouse. It would then run the right set of tests (with `go test`).
- I don't want to make the final decision here, but I'd rather not go with the current approach if possible. If it helps, you could split up this PR and have the test cases and framework in a separate PR. We could then merge this PR first.

Reply:
let's move e2e to a follow-up PR. don't think we have enough time for 1.6 for the change and reviews.